core/src/main/java/org/jdbi/v3/core/mapper/reflect/internal/PojoMapper.java

```diff
@@ -0,0 +1,174 @@
+                                               List<ColumnNameMatcher> columnNameMatchers,
+                                               List<String> unmatchedColumns) {
+        final List<RowMapper<?>> mappers = new ArrayList<>();
+        final List<PojoProperty<T>> propList = new ArrayList<>();
+
+        for (PojoProperty<T> property : properties.getProperties().values()) {
+            Nested anno = property.getAnnotation(Nested.class).orElse(null);
+
+            if (anno == null) {
+                String paramName = prefix + getName(property);
+
+                findColumnIndex(paramName, columnNames, columnNameMatchers, () -> debugName(property))
+                    .ifPresent(index -> {
+                        @SuppressWarnings({ "unchecked", "rawtypes" })
+                        ColumnMapper<?> mapper = ctx.findColumnMapperFor(property.getQualifiedType())
+                            .orElseGet(() -> (ColumnMapper) defaultColumnMapper(property));
+
+                        mappers.add(new SingleColumnMapper<>(mapper, index + 1));
+                        propList.add(property);
+
+                        unmatchedColumns.remove(columnNames.get(index));
+                    });
+            } else {
+                String nestedPrefix = prefix + anno.value();
+                if (anyColumnsStartWithPrefix(columnNames, nestedPrefix, columnNameMatchers)) {
+                    nestedMappers
+                        .computeIfAbsent(property, d -> createNestedMapper(ctx, d, nestedPrefix))
+                        .specialize0(ctx, columnNames, columnNameMatchers, unmatchedColumns)
+                        .ifPresent(nestedMapper -> {
+                            mappers.add(nestedMapper);
+                            propList.add(property);
+                        });
+                }
+            }
+        }
+
+        if (mappers.isEmpty() && !columnNames.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return Optional.of((r, c) -> {
+            final PojoBuilder<T> pojo = properties.create();
+
+            for (int i = 0; i < mappers.size(); i++) {
+                RowMapper<?> mapper = mappers.get(i);
+                PojoProperty<T> property = propList.get(i);
+
+                Object value = mapper.map(r, ctx);
+
+                pojo.set(property, value);
+            }
+
+            return pojo.build();
+        });
+    }
+
+    @SuppressWarnings({ "rawtypes", "unchecked" }) // Type loses <T>
+    protected PojoMapper<?> createNestedMapper(StatementContext ctx, PojoProperty<T> property, String nestedPrefix) {
+        final Type propertyType = property.getQualifiedType().getType();
+        return new PojoMapper(
+                GenericTypes.getErasedType(propertyType),
+                ctx.getConfig(PojoTypes.class).propertiesOf(type)
+                    .orElseThrow(() -> new UnableToProduceResultException("Couldn't find properties for nested type " + propertyType, ctx)),
```

Comment:
Whoa, does this mean we can mix and match pojo types for nested mappers?

Reply:
I'd expect no less from a recursive system tbh. Cool that it's now implemented. :+1:

Reply:
Yup!  Eventually this could be a more flexible replacement for the entire existing bean binding infrastructure.